### ISSUE TITLE
Unfortunate line splitting behaviour

### ISSUE THREAD

**Issue by jdm** [rust-lang-nursery/rustfmt#286]
```diff
-        let cx = tp1.x + any * radius * if anticlockwise { 1.0 } else { -1.0 };
-        let cy = tp1.y - anx * radius * if anticlockwise { 1.0 } else { -1.0 };
+        let cx = tp1.x +
+                 any * radius *
+                 if anticlockwise {
+            1.0
+        } else {
+            -1.0
+        };
+        let cy = tp1.y -
+                 anx * radius *
+                 if anticlockwise {
+            1.0
+        } else {
+            -1.0
+        };
```

**Comment by marcusklaas** [comment]
There is the `single_line_if_else` config bool, which could help here. Otherwise, how should we format this?

**Comment by jdm** [comment]
I guess it's not clear to me why we end up with `any * radius` on one line and other things with one item per line? The formatting of the `if`/`else` blocks probably makes this look worse than it ordinarily would, too.

**Comment by marcusklaas** [comment]
It's an oddity of the heuristic we're using. We're formatting the expression in the order that they are laid out in the AST. Addition has the lowest precedence, so that's why we see a weird split there. It needs tweaking.

**Comment by nrc** [comment]
Also the indentation of the if block is pretty sad.

**Comment by nrc** [comment]
Closed by #1017
